### PR TITLE
enterprise-gateway-2.3.0

### PR DESCRIPTION
**Pull request by ntfrnzn**
reference
https://jupyter-enterprise-gateway.readthedocs.io/en/latest/kernel-kubernetes.html#configuration

modified from
https://github.com/jupyter/enterprise_gateway/releases/download/v2.3.0/jupyter_enterprise_gateway_helm-2.3.0.tgz

### CODE REVIEW

**Comment by ntfrnzn** [comment]
@liranp this is one of the wave components we need to host a chart for since
(a) it's not being hosting in a chart repo at all
(b) we have to make some modifications for the notebook namespace

@thorsteinnth I tested with and without a specified namespace, but please take a look

**Comment by ntfrnzn** [comment]
I don't know what this linter is expecting, according to https://helm.sh/docs/topics/charts/#the-chartyaml-file, maintainers.name is just a name, not a url.

File: charts/enterprise-gateway/templates/notebooks.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  namespace: {{ .Values.kernel.kernelNamespace }}
+  labels:
+    app: enterprise-gateway
+    component: kernel
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+subjects:
+  - kind: ServiceAccount
+    name: default
```
Comment (thorsteinnth) [inline charts/enterprise-gateway/templates/notebooks.yaml:38]: do we want to have the default service account here too? or enforce all notebooks use `kernel-sa`

File: charts/enterprise-gateway/values.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# Master public IP on which to expose EG.
+k8sMasterPublicIP:
+
+kernel:
+  # Kernel cluster role created by this chart.
+  clusterRole: kernel-controller
+  # Will start kernels in the same namespace as EG if True.
+  shareGatewayNamespace: false
+  # ignored if shareGatewayNamespace
+  kernelNamespace: spark-notebooks
```
Comment (thorsteinnth) [inline charts/enterprise-gateway/values.yaml:22]: Currently we also have non-Spark kernels avaiable to the user (vanilla python, vanilla R, vanilla Scala, and Python with Tensorflow).  I think we should just call the namespace `notebooks`.

Perhaps we should restrict the available kernels to only Spark kernels, I'm not sure (the `whitelist` configuration value below).

**Comment by ntfrnzn** [comment]
fwiw, with the default chart schema, the maintainer must be a valid github account, https://github.com/helm/chart-testing/issues/192 ¯\_(ツ)_/¯

**Review by thorsteinnth — APPROVED** [review]

**Review by liranp — APPROVED** [review]

**Pull request merged**